AppKit/CPView.j

```diff
@@ -558,7 +560,7 @@ var CPViewHighDPIDrawingEnabled = YES;
 #endif
 
     // Notify the subview that it will be moving.
-    [aSubview viewWillMoveToSuperview:self];
+    [aSubviewOrViewAnimator viewWillMoveToSuperview:self];
```

Comment:
You mean sure it should be supported to add an animator, or the way it is implemented. For me it seems logical that the special event keypath CPAnimationTriggerOrderIn is invoked when the animator is added to the superview : [view addSubview:animator] . Although, to be honest, this is not working in cocoa (it crashes) but I can't say what is the right way to reach this key path because it seems it has been silently deprecated. Or I missed something.

For the implementation of this feature, I can think of a more elegant way to do it: the animator could register for viewWill/DidAppear notifications sent by its target (the actual view) and perform the animation there.